### ISSUE TITLE
fastertransformer nan error

### ISSUE THREAD

**Issue by leerumor** [NVIDIA/FasterTransformer#43]
When using my own pretrained model, I got NaN from the eight transformer layer.
By reading "fastertransformer/cuda/open_attention.cu", I found that the softmax input is not protected (substract the max value), the right formula shoud be:
```
def softmax(x):
    """Compute softmax values for each sets of scores in x."""
    e_x = np.exp(x - np.max(x))
    return e_x / e_x.sum()
```
I'm trying to modify "softmax_kernel_v2", maybe someone can offer a better version
